src/Core/Commander/Providers/BuildingBox_Provider.js

```diff
@@ -99,7 +98,8 @@ BuildingBox_Provider.prototype.generateMesh = function generateMesh(elements, bb
                 const vector3_1 = new THREE.Vector3(pgeo1.x, pgeo1.y, pgeo1.z); // - x temporary, bug
                 const vector3_2 = new THREE.Vector3(pgeo2.x, pgeo2.y, pgeo2.z);
 
-                arrPoint2D.push(CVML.newPoint(p1.z, p1.x)); // -pgeo1.x, pgeo1.z)); //for roof
+
+                arrPoint2D.push(p1.z, p1.x);
```

Comment:
Why not simply make `p1` a Vector2 and doing `arrPoint2D.push(p1)`?
It would simplify the loop code below to:
```js
    const pt1 = arrPoint2D[triangles[w]];
    const pt2 = arrPoint2D[triangles[w + 1]];
    const pt3 = arrPoint2D[triangles[w + 2]];
```
Or even:
```js
    const [pt1, pt2, pt3] = arrPoint2D.slice(w, w+3);
```